File: Presentation Layer (frontend)/Reservation_angular/src/app/service/sysImage.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { MenuItemReview } from "src/data/menuItemReview";
import { OrderItem } from 'src/data/orderItem';
import { SysImage } from 'src/data/sysImage';
@Injectable({
  providedIn: 'root'
})
export class SysImageService {

  headerDict={
    'Content-Type':'application/json',
    'Accept':'application/json'
    }
 requestOptions={
      headers:new HttpHeaders(this.headerDict)
      }
 constructor(public http:HttpClient,private route:Router) { }
AddSysImage(sysImage:SysImage){
    
}
UpdateSysImage(sysImage:SysImage){}
DeleteSysImage(id:number){}
GetAllSysImages(){}
GetSysImagesByName(name:string){}
}
    

File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/SysImageService.cs
using Reservation.Core.Data;
using Reservation.Core.Repository;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservation.Infra.Service
{
    public class SysImageService: ISysImageService
    {
        private readonly ISysImageRepository _sysImageRepository;
        public SysImageService(ISysImageRepository sysImageRepository)
        {
            _sysImageRepository = sysImageRepository;
        }

        public bool AddSysImage(SysImage sysImage)
        {
            return _sysImageRepository.AddSysImage(sysImage);
        }

        public bool DeleteSysImage(int id)
        {
            return _sysImageRepository.DeleteSysImage(id);
        }

        public List<SysImage> GetAllSysImages()
        {
            return _sysImageRepository.GetAllSysImages();
        }

        public List<SysImage> GetSysImagesByName(string name)
        {
            return _sysImageRepository.GetSysImagesByName(name);
        }

        public bool UpdateSysImage(SysImage sysImage)
        {
            return _sysImageRepository.UpdateSysImage(sysImage);
        }
    }
}
